wherehows-web/app/components/hotkeys/global-hotkeys.ts

```diff
@@ -68,15 +73,15 @@ export default class GlobalHotkeys extends Component {
     const body = document.querySelector('.ember-application');
 
     if (body) {
-      body.addEventListener('keyup', this.onKeyUp.bind(this));
+      body.addEventListener('keyup', set(this, 'bodyEventListener', this.onKeyUp.bind(this)));
```

Comment:
@cptran777  sorry my comment before was terse without a detailed explanation, had to take a phone call.
I believe you could do and not need to use set
```ts
  
  onKeyUp = () => { ... };

  didInsertElement() {
    document.querySelector(...).addEventListener('keyup', this.onKeyUp);
  }

  willDestoryElement() {
    document.querySelector(...).removeEventListener('keyup', this.onKeyUp);
  }
```

Reply:
Not sure if that is going to work but I prefer this:
```js
document.querySelector(...).removeEventListener('keyup', (e) => this.onKeyUp(e));
```

Reply:
will not work

Reply:
yep, but yours neither :P

Reply:
We have to bind keyup because it loses the this context when passed into the event listener

Reply:
wait let me think...

if it is a class, onKeyUp default value will happen in the constructor... so `this` is assigned... oh... it will work... damn Seyi you are always right!

Reply:
@theseyi ^

Reply:
I just tried it, it works perfectly
```ts
  onKeyUp = () => {
    alert('Hello');
  };

  didInsertElement() {
    document.body.addEventListener('click', this.onKeyUp);
    setTimeout(() => document.body.removeEventListener('click', this.onKeyUp), 5000);
  }

  willDestoryElement() {
    document.body.removeEventListener('click', this.onKeyUp);
  }
```

Reply:
@theseyi It works if onKeyUp doesn't have any this context inside it. 
But our use case is:
```
  onKeyUp(e: KeyboardEvent) {
    const target = <HTMLElement>e.target;

    if (this.isEligibleTarget(target)) {
      get(this, 'hotKeys').applyKeyMapping(e.keyCode);
    }
  }
```
That this has to mean the class. But if you don't use this.onKeyUp.bind(this), then the this context I think gets set to the addEventListener function or window (can't remember) when this.onKeyUp gets run in response to the event.

Reply:
Sorry, the `this` context is the element that fired the event. In the above case `this.isElligibleTarget() is not a function` will be an error.